pkg/services/serviceaccounts/database/store.go

```diff
@@ -412,27 +449,28 @@ func (s *ServiceAccountsStoreImpl) MigrateApiKeysToServiceAccounts(ctx context.C
 	return migrationResult, nil
 }
 
-func (s *ServiceAccountsStoreImpl) MigrateApiKey(ctx context.Context, orgId int64, keyId int64) error {
+func (s *ServiceAccountsStoreImpl) MigrateApiKey(ctx context.Context, orgId int64, keyId int64) (int64, error) {
 	basicKeys, err := s.apiKeyService.GetAllAPIKeys(ctx, orgId)
 	if err != nil {
-		return err
+		return 0, err
 	}
 	if len(basicKeys) == 0 {
-		return fmt.Errorf("no API keys to convert found")
+		return 0, fmt.Errorf("no API keys to convert found")
 	}
 	for _, key := range basicKeys {
 		if keyId == key.ID {
-			err := s.CreateServiceAccountFromApikey(ctx, key)
+			serviceAccountID, err := s.CreateServiceAccountFromApikey(ctx, key)
 			if err != nil {
 				s.log.Error("Converting to service account failed with error", "keyId", keyId, "error", err)
-				return err
+				return 0, err
 			}
+			return serviceAccountID, nil
 		}
 	}
-	return nil
+	return 0, fmt.Errorf("no API keys to convert found")
```

Comment:
let's not change behaviour of this method and create another one instead

Reply:
No changes in MigrateApiKey anymore.